components/collector/src/source_collectors/api_source_collectors/azure_devops/merge_requests.py

```diff
@@ -0,0 +1,66 @@
+"""Azure Devops Server merge requests collector."""
+
+from typing import cast
+
+from collector_utilities.functions import match_string_or_regular_expression
+from collector_utilities.type import URL
+from source_model import Entity, SourceMeasurement, SourceResponses
+
+from .base import AzureDevopsRepositoryBase
+
+
+class AzureDevopsMergeRequests(AzureDevopsRepositoryBase):
+    """Collector for merge requests (pull requests in Azure Devops)."""
+
+    async def _api_url(self) -> URL:
+        """Extend to add the pull requests API path."""
+        api_url = str(await super()._api_url())
+        return URL(f"{api_url}/pullrequests?api-version=4.1&searchCriteria.status=all")
+
+    async def _landing_url(self, responses: SourceResponses) -> URL:
+        """Extend to add the pull requests path."""
+        landing_url = str(await super()._landing_url(responses))
+        return URL(f"{landing_url}/pullrequests")
+
+    async def _parse_source_responses(self, responses: SourceResponses) -> SourceMeasurement:
+        """Override to parse the merge requests from the responses."""
+        merge_requests = []
+        for response in responses:
+            merge_requests.extend((await response.json())["value"])
+        landing_url = (await self._landing_url(responses)).rstrip("s")
+        entities = [
+            Entity(
+                key=merge_request["pullRequestId"],
+                title=merge_request["title"],
+                target_branch=merge_request["targetRefName"],
+                url=f"{landing_url}/{merge_request['pullRequestId']}",
+                state=merge_request["status"],
+                created=merge_request.get("creationDate"),
+                closed=merge_request.get("closedDate"),
+                downvotes=str(self._downvotes(merge_request)),
+                upvotes=str(self._upvotes(merge_request)),
+            )
+            for merge_request in merge_requests
+            if self._include_merge_request(merge_request)
+        ]
+        return SourceMeasurement(entities=entities, total=str(len(merge_requests)))
+
+    def _include_merge_request(self, merge_request) -> bool:
+        """Return whether the merge request should be counted."""
+        min_upvotes = int(cast(str, self._parameter("upvotes")))
+        request_has_fewer_than_min_upvotes = min_upvotes == 0 or self._upvotes(merge_request) < min_upvotes
```

Comment:
Is there a reason that this will always be true if the minimum amount of upvotes is 0? Shouldn't it always be false in that case?

Reply:
It's a bit counterintuitive so I'll add a comment, but I believe it's correct as currently implemented: if the minimum number of upvotes is zero, merge requests are included regardless of how many upvotes they actually have. If the minimum number of upvotes is more than zero then only merge requests that have less than the minimum number of upvotes are included in the count.